src/node/history.h

```diff
@@ -490,13 +500,80 @@ namespace ccf
       NodeId id_,
       tls::KeyPair& kp_,
       Signatures& sig_,
-      Nodes& nodes_) :
+      Nodes& nodes_,
+      size_t sig_tx_interval_ = 0,
+      size_t sig_ms_interval_ = 0) :
       store(store_),
       id(id_),
       kp(kp_),
       signatures(sig_),
-      nodes(nodes_)
-    {}
+      nodes(nodes_),
+      sig_tx_interval(sig_tx_interval_),
+      sig_ms_interval(sig_ms_interval_)
+    {
+      start_signature_emit_timer();
+    }
+
+    void start_signature_emit_timer()
+    {
+      struct EmitSigMsg
+      {
+        EmitSigMsg(HashedTxHistory<T>* self_) : self(self_) {}
+        HashedTxHistory<T>* self;
+      };
+
+      auto emit_sig_msg = std::make_unique<threading::Tmsg<EmitSigMsg>>(
+        [](std::unique_ptr<threading::Tmsg<EmitSigMsg>> msg) {
+          auto self = msg->data.self;
+
+          std::unique_lock<SpinLock> mguard(
+            self->signature_lock, std::defer_lock);
+
+          const int64_t sig_ms_interval = self->sig_ms_interval;
+          int64_t time_since_last_sig = sig_ms_interval;
+
+          if (mguard.try_lock())
+          {
+            auto time = threading::ThreadMessaging::thread_messaging
+                          .get_current_time_offset()
+                          .count();
+            auto time_of_last_signature = self->time_of_last_signature.count();
+
+            auto consensus = self->store.get_consensus();
+            if (
+              (consensus != nullptr) && consensus->is_primary() &&
+              self->store.commit_gap() > 0 && time > time_of_last_signature &&
```

Comment:
I still think that `(time - time_of_last_signature) > sig_ms_interval` is stronger than `time > time_of_last_signature` and so we remove the latter for clarity.